### ISSUE TITLE
Rename `no-array-for-each` to `no-for-each`?

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1778]
It's actually check other types, `Set` (Buggy #1452)/`Map`/`NodeList`

**Comment by sindresorhus** [comment]
Makes sense, but not sure it's worth the effort or churn. Maybe we should just collect it in a list and do multiple renames in a batch when we have some more? As a user, it's better to handle multiple deprecations in one go instead of multiple times.

**Comment by fisker** [comment]
Another rule rename idea, `prevent-abbreviations` should renamed as `no-abbreviations`.

**Comment by fregante** [comment]
> `prevent-abbreviations` should renamed as `no-abbreviations`

As the description suggests, the rule doesn't necessarily only work on abbreviations, so maybe it should be called `text-replacements` (it'd match macOS/iOS naming)

**Comment by fisker** [comment]
Deprecated `prefer-node-protocol` , since it's adopted by `eslint-plugin-n` https://github.com/eslint-community/eslint-plugin-n/blob/master/docs/rules/prefer-node-protocol.md

**Comment by sindresorhus** [comment]
> As the description suggests, the rule doesn't necessarily only work on abbreviations, so maybe it should be called text-replacements (it'd match macOS/iOS naming)

I kinda like the name `name-replacements`. Thoughts?

Since this rule isn't really about abbreviations. It's just that every default replacement happens to be one, but the engine is a general. That rules out `no-abbreviations`.

`text-replacements` is recognizable but inaccurate. The rule never touches raw text; no string contents, no comments. It does renaming of *names*: variables, properties, exported names, and filenames.

`name-replacements` fixes both: it's accurate ("name" is the exact superset of what's touched, including filenames, which "identifier" would exclude)

**Comment by sindresorhus** [comment]
> Deprecate prefer-node-protocol , since it's adopted by eslint-plugin-n https://github.com/eslint-community/eslint-plugin-n/blob/master/docs/rules/prefer-node-protocol.md

While we won't add more Node.js specific plugins, I don't see the value of churning this by deprecating it. I'm also unsure about the future maintainance of the `n` plugin. It has been spotty in the past.

**Comment by sindresorhus** [comment]
> Rename no-array-for-each to no-for-each.

Done

**Comment by fregante** [comment]
> name-replacements

Technically `identifier-replacements` but `name-replacements` is easier

**Comment by sindresorhus** [comment]
> Technically identifier-replacements but name-replacements is easier

The rule also works for filenames. And even ignoring that, "name" is a more user friendly term.
